### ISSUE TITLE
About header settings and corsHandler. Is this a bug?

### ISSUE THREAD

**Issue by myaaaapon** [Baroshem/nuxt-security#505]
Thanks for developing this useful module!! I always appreciate your developments 😊
I have three questions and suggestions.

```
node v20.16.0
nuxt@3.12.4
nuxt-security@2.0.0-rc.9

```

# 1, Information on js/css default values in the documentation.
It seems that the header values are different for HTML and the rest of the files.
I have spent a lot of time mistaking this for a bug.
I think it would be more helpful to mention this in the documentation

・[Defaults](https://nuxt-security.vercel.app/documentation/getting-started/configuration#defaults)
・Mentioned below. #434

# 2, The header settings under /server/api are the same as js/css. Is this a bug?

Mentioned below. #434

# 3, corsHandler settings are not reflected. Is this a bug?

I want to set `Access-Control-Allow-Origin: http://localhost:3000`.

If it is as follows, it is `not` reflected.
```nuxt.config.ts
# nuxt.config.ts
# This is a problem.

export default {
    routeRules: {
        '/api/**': {
            security: {
                corsHandler: {
                    origin: 'http://localhost:3000'
                },
            },
        }
    },
}
```

It will be reflected if you do the following.
```nuxt.config.ts
# nuxt.config.ts
# OK

export default {
    routeRules: {
        '/api/**': {
            security: {
                corsHandler: false
            },
            headers: {
                'Access-Control-Allow-Origin': 'http://localhost:3000'
            }
        }
    },
}
```
I would like default values for methods other than `origin`, `allowHeaders` and other items.
In other words, how can I make only `origin` a unique setting?
- [CORS Handler](https://nuxt-security.vercel.app/documentation/middleware/cors-handler#cors-handler)

**Comment by vejja** [comment]
Hi @myaaaapon 

You are right that we do not apply the same settings to HTML resources, JS/CSS files and backend routes.
This is documented here : https://nuxt-security.vercel.app/documentation/advanced/security-headers
Maybe the documentation is not clear enough. Let us know how if you have suggestions for reorganizing the docs. 

The logic is basically the following :
- some security headers are only valid for HTML resources (e.g. Content Security Policy, Permissions Policy)
- all other security headers are valid for all resources (e.g Strict Transport Security), so this includes HTML, JS/CSS, and backend routes
- middlewares only apply to backend routes

JS/CSS files (and all other static files) are delivered differently whether you are using SSR (delivered by the Nuxt server) or SSG (delivered by your CDN or your custom static server). If you have difficulty understanding the headers of your JS/CSS files, let me know your server setup so that I can further guide you.


As far as the CORS handler is concerned, I’ll have a look at how we resolve the rules and let you know here
Cheers

**Comment by Baroshem** [comment]
Hey @myaaaapon any feedback from your side? Is there anything we could do to improve the documenation? :)

**Comment by myaaaapon** [comment]
Hi @vejja 

Thank you for your detailed answer 😊

I had missed your advanced settings...
- [Security Headers](https://nuxt-security.vercel.app/documentation/advanced/security-headers#security-headers)

This gave me a good understanding of the difference between HTML and the headers of the rest of the file.
It helped me a lot. Thank you.

If any policy on CORS handlers has been decided on your end,
I would be grateful if you could share it with me.

----

Hi @Baroshem 
Maybe, like me, "The default header values are not set!" and someone might mistake it as a defect, I felt it would be easier to understand if there was a link or something in the "default configuration" page.
- [Configration](https://nuxt-security.vercel.app/documentation/getting-started/configuration#defaults)
- [Security Headers](https://nuxt-security.vercel.app/documentation/advanced/security-headers#security-headers)

## addition

I found an mistake in the documentation.
- URL: [Allowed HTTP Methods](https://nuxt-security.vercel.app/documentation/middleware/allowed-methods-restricter#allowed-http-methods)

```diff
- type HTTPMethod = 'GET' | 'POST' | 'DELETE' | 'PATCH' | 'POST' | string;

# Perhaps the following is correct.
+ type HTTPMethod = 'GET' | 'POST' | 'DELETE' | 'PATCH' | 'PUT' | string;
```

@vejja @Baroshem 
Finally, thank you for your very polite and prompt reply. I used to be in trouble and you have helped me a lot 😊

**Comment by Baroshem** [comment]
@myaaaapon Thanks for the extensive research from your side!

Would you be interested in contributing to the project with the bugfixes you mentioned? I will be more than happy to help you :)

**Comment by Baroshem** [comment]
Hey @myaaaapon bumping the topic above :)
